src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -221,7 +223,9 @@ public Builder addMember(String name, CodeBlock codeBlock) {
      * depending on the given {@code value} object. Falls back to {@code "$L"} literal format if
      * the class of the given {@code value} object is not supported.
      */
-    private Builder addValue(String memberName, Object value) {
+    public Builder addMemberForValue(String memberName, Object value) {
+      checkArgument(memberName != null, "memberName == null");
+      checkArgument(value != null, "only constant non-null values are permitted (%s)", memberName);
```

Comment:
Or introduce a `Util.checkArgumentNotNull(memberName, "memberName")`? That a) checks for `null` and b) prepares a `argument memberName must not be null` and c-1) throws a `IllegalArgumentException` or c-2) a `NullPointerException`.